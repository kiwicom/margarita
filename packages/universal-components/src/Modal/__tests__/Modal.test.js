// @flow

import * as React from 'react';
import { render, fireEvent } from 'react-native-testing-library';

import { Text } from '../..';
import ModalNative from '../Modal.native';
import ModalWeb from '../Modal.web';

const placeholder = 'Modal test content';

describe('Modal - web', () => {
  const onBackdropPress = jest.fn();
  const { getByText, getByTestId } = render(
    <ModalWeb isVisible onBackdropPress={onBackdropPress}>
      <Text>{placeholder}</Text>
    </ModalWeb>,
  );

  it('should contain passed children', () => {
    expect(getByText(placeholder)).toBeDefined();
  });

  it('should execute onBackdropPress method', () => {
    fireEvent(getByTestId('backdrop'), 'press');
    expect(onBackdropPress).toHaveBeenCalledWith(undefined);
  });

  it('should match snapshot', () => {
    const component = render(
      <ModalWeb isVisible onBackdropPress={jest.fn()}>
        <Text>{placeholder}</Text>
      </ModalWeb>,
    );

    expect(component).toMatchSnapshot();
  });
});

describe('Modal - native', () => {
  it('should match snapshot', () => {
    const component = render(
      <ModalNative isVisible={false} onBackdropPress={jest.fn()}>
        <Text>{placeholder}</Text>
      </ModalNative>,
    );

    expect(component).toMatchSnapshot();
  });
});
