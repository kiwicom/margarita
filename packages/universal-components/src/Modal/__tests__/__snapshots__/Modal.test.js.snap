// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal - native should match snapshot 1`] = `
<Modal
  animationType="none"
  hardwareAccelerated={false}
  hideModalContentWhileAnimating={true}
  onModalHide={[Function]}
  onRequestClose={[Function]}
  scrollOffset={0}
  scrollOffsetMax={0}
  scrollTo={null}
  supportedOrientations={
    Array [
      "portrait",
      "landscape",
    ]
  }
  swipeThreshold={100}
  transparent={true}
  visible={false}
>
  <View
    accessible={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "black",
        "bottom": 0,
        "height": 1334,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 750,
      }
    }
  />
  <View
    hideModalContentWhileAnimating={true}
    onModalHide={[Function]}
    pointerEvents="box-none"
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
        "margin": 37.5,
        "transform": Array [
          Object {
            "translateY": 0,
          },
        ],
      }
    }
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": "Roboto",
            "margin": 0,
          },
        ]
      }
    >
      Modal test content
    </Text>
  </View>
</Modal>
`;

exports[`Modal - web should match snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "height": "100%",
        "justifyContent": "center",
        "start": 0,
        "top": 0,
        "width": "100%",
        "zIndex": 825,
      },
      undefined,
    ]
  }
>
  <View
    accessible={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "height": "100%",
          "width": "100%",
        },
        Object {
          "backgroundColor": "#171b1e",
          "opacity": 0.5,
        },
      ]
    }
    testID="backdrop"
  />
  <Text
    style={
      Array [
        Object {
          "fontFamily": "Roboto",
          "margin": 0,
        },
      ]
    }
  >
    Modal test content
  </Text>
</View>
`;
